sway/desktop/xwayland.c

```diff
@@ -121,6 +121,22 @@ static void unmanaged_handle_unmap(struct wl_listener *listener, void *data) {
 	}
 }
 
+static void unmanaged_handle_request_activate(struct wl_listener *listener, void *data) {
+	struct wlr_xwayland_surface *xsurface = data;
+	if (!xsurface->mapped) {
+		return;
+	}
+	struct sway_seat *seat = input_manager_current_seat();
+	struct sway_container *focus = seat_get_focused_container(seat);
+	if (focus && focus->view->pid != xsurface->pid) {
```

Comment:
It's possible that the current focus isn't a view. We should check for `focus->view` here as well.